windows-terminal/config/themes/new-rebels.omp.json

```diff
@@ -344,38 +344,50 @@
                     "foreground": "#cccc00",
                     "invert_powerline": true,
                     "powerline_symbol": "\ue0b2",
-                    "properties": {
-                        "template": " \uf21b "
-                    },
                     "style": "powerline",
+                    "template": " \uf21b ",
                     "type": "root"
                 },
                 {
                     "background": "#ff8080",
                     "foreground": "#ffffff",
                     "powerline_symbol": "\ue0b0",
-                    "properties": {
-                        "template": " \ue20f {{ if gt .Code 0 }}\uf00d {{ .Meaning }}{{ else }}\uf42e{{ end }} "
-                    },
                     "style": "powerline",
+                    "template": " \ue20f {{ if gt .Code 0 }}\uf00d {{ .Meaning }}{{ else }}\uf42e{{ end }} ",
                     "type": "exit"
                 },
                 {
                     "background": "#4075CC",
                     "foreground": "transparent",
                     "powerline_symbol": "\ue0b0",
-                    "properties": {
-                        "template": " {{ or .Env.OHMYPOSH_SHELL_LOGO \"\uf1d0\" }}  \uf054"
-                    },
                     "style": "powerline",
+                    "foreground_templates": [
+                        "{{if eq .Name \"bash\" }}#00ff00{{end}}",
+                        "{{if eq .Name \"zsh\" }}#0000ff{{end}}",
+                        "{{if eq .Name \"pwsh\" }}#ff7979{{end}}"
+                    ],
+                    "background_templates": [
+                        "{{if eq .Name \"bash\" }}#228b22{{end}}",
+                        "{{if eq .Name \"zsh\" }}#4169e1{{end}}",
+                        "{{if eq .Name \"pwsh\" }}#52205f{{end}}"
+                    ],
+                    "template": " {{ or .Env.OHMYPOSH_SHELL_LOGO \"\uf1d0\" }}  \uf054",
                     "type": "shell"
                 },
                 {
                     "foreground": "#4075CC",
-                    "properties": {
-                        "template": "\ue285\uf641"
-                    },
                     "style": "plain",
+                    "foreground_templates": [
+                        "{{if eq .Shell \"bash\" }}#228b22{{end}}",
```

Comment:
Should it be `.Name` instead of `.Shell`?

(Same for the other 2 lines)

Reply:
Well, in the "text" section type, it is only possible to know the current shell name using ".Shell" (so there was an issue at background_template) thanks for enlightening it :)... And, in the "shell" section type, because it knows that a shell, it only required .Name...